File: SafeDesk365.SDK/Api/ApiRequestBuilder.cs
using Microsoft.Kiota.Abstractions;
using SafeDesk365.SDK.Api.Bookings;
using SafeDesk365.SDK.Api.DeskAvailabilities;
using SafeDesk365.SDK.Api.Desks;
using SafeDesk365.SDK.Api.Facilities;
using SafeDesk365.SDK.Api.Locations;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
namespace SafeDesk365.SDK.Api {
    /// <summary>Builds and executes requests for operations under \api</summary>
    public class ApiRequestBuilder {
        /// <summary>The bookings property</summary>
        public BookingsRequestBuilder Bookings { get =>
            new BookingsRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The deskAvailabilities property</summary>
        public DeskAvailabilitiesRequestBuilder DeskAvailabilities { get =>
            new DeskAvailabilitiesRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The desks property</summary>
        public DesksRequestBuilder Desks { get =>
            new DesksRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The facilities property</summary>
        public FacilitiesRequestBuilder Facilities { get =>
            new FacilitiesRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The locations property</summary>
        public LocationsRequestBuilder Locations { get =>
            new LocationsRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>Path parameters for the request</summary>
        private Dictionary<string, object> PathParameters { get; set; }
        /// <summary>The request adapter to use to execute the requests.</summary>
        private IRequestAdapter RequestAdapter { get; set; }
        /// <summary>Url template to use to build the URL for the current request builder</summary>
        private string UrlTemplate { get; set; }
        /// <summary>
        /// Instantiates a new ApiRequestBuilder and sets the default values.
        /// <param name="pathParameters">Path parameters for the request</param>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        /// </summary>
        public ApiRequestBuilder(Dictionary<string, object> pathParameters, IRequestAdapter requestAdapter) {
            _ = pathParameters ?? throw new ArgumentNullException(nameof(pathParameters));
            _ = requestAdapter ?? throw new ArgumentNullException(nameof(requestAdapter));
            UrlTemplate = "{+baseurl}/api";
            var urlTplParams = new Dictionary<string, object>(pathParameters);
            PathParameters = urlTplParams;
            RequestAdapter = requestAdapter;
        }
        /// <summary>
        /// Instantiates a new ApiRequestBuilder and sets the default values.
        /// <param name="rawUrl">The raw URL to use for the request builder.</param>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        /// </summary>
        public ApiRequestBuilder(string rawUrl, IRequestAdapter requestAdapter) {
            if(string.IsNullOrEmpty(rawUrl)) throw new ArgumentNullException(nameof(rawUrl));
            _ = requestAdapter ?? throw new ArgumentNullException(nameof(requestAdapter));
            UrlTemplate = "{+baseurl}/api";
            var urlTplParams = new Dictionary<string, object>();
            urlTplParams.Add("request-raw-url", rawUrl);
            PathParameters = urlTplParams;
            RequestAdapter = requestAdapter;
        }
    }
}

File: SDK/NPM/api/apiRequestBuilder.ts
import {BookingsRequestBuilder} from './bookings/bookingsRequestBuilder';
import {BookingsItemRequestBuilder} from './bookings/item/bookingsItemRequestBuilder';
import {DeskAvailabilitiesRequestBuilder} from './deskAvailabilities/deskAvailabilitiesRequestBuilder';
import {DesksRequestBuilder} from './desks/desksRequestBuilder';
import {FacilitiesRequestBuilder} from './facilities/facilitiesRequestBuilder';
import {LocationsRequestBuilder} from './locations/locationsRequestBuilder';
import {getPathParameters, RequestAdapter} from '@microsoft/kiota-abstractions';

/** Builds and executes requests for operations under /api  */
export class ApiRequestBuilder {
    /** The bookings property  */
    public get bookings(): BookingsRequestBuilder {
        return new BookingsRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The deskAvailabilities property  */
    public get deskAvailabilities(): DeskAvailabilitiesRequestBuilder {
        return new DeskAvailabilitiesRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The desks property  */
    public get desks(): DesksRequestBuilder {
        return new DesksRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The facilities property  */
    public get facilities(): FacilitiesRequestBuilder {
        return new FacilitiesRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The locations property  */
    public get locations(): LocationsRequestBuilder {
        return new LocationsRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** Path parameters for the request  */
    private readonly pathParameters: Record<string, unknown>;
    /** The request adapter to use to execute the requests.  */
    private readonly requestAdapter: RequestAdapter;
    /** Url template to use to build the URL for the current request builder  */
    private readonly urlTemplate: string;
    /**
     * Gets an item from the SafeDesk.API.api.bookings.item collection
     * @param id Unique identifier of the item
     * @returns a BookingsItemRequestBuilder
     */
    public bookingsById(id: string) : BookingsItemRequestBuilder {
        if(!id) throw new Error("id cannot be undefined");
        const urlTplParams = getPathParameters(this.pathParameters);
        urlTplParams["id"] = id
        return new BookingsItemRequestBuilder(urlTplParams, this.requestAdapter);
    };
    /**
     * Instantiates a new ApiRequestBuilder and sets the default values.
     * @param pathParameters The raw url or the Url template parameters for the request.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public constructor(pathParameters: Record<string, unknown> | string | undefined, requestAdapter: RequestAdapter) {
        if(!pathParameters) throw new Error("pathParameters cannot be undefined");
        if(!requestAdapter) throw new Error("requestAdapter cannot be undefined");
        this.urlTemplate = "{+baseurl}/api";
        const urlTplParams = getPathParameters(pathParameters);
        this.pathParameters = urlTplParams;
        this.requestAdapter = requestAdapter;
    };
}
